www/caddy/src/opnsense/mvc/app/controllers/OPNsense/Caddy/Api/ReverseProxyController.php

```diff
@@ -39,24 +39,69 @@ class ReverseProxyController extends ApiMutableModelControllerBase
     protected static $internalModelClass = 'OPNsense\Caddy\Caddy';
     protected static $internalModelUseSafeDelete = true;
 
-    /*ReverseProxy Section*/
-
-    /*Search Function adjusted for the search filter dropdown*/
-    public function searchReverseProxyAction()
+    /**
+     * Function for search filter dropdown
+     *
+     * @return array containing rows of domain and port combinations.
+     */
+    public function getAllReverseDomainsAction()
     {
-        // Get a comma-separated list of UUIDs from the request
-        $reverseUuids = $this->request->get('reverseUuids');
+        $this->sessionClose(); // Close session early for performance
+        $result = array("rows" => array());
+
+        $mdlCaddy = new \OPNsense\Caddy\Caddy();
+        $reverseNodes = $mdlCaddy->reverseproxy->reverse->iterateItems();
+
+        foreach ($reverseNodes as $item) {
+            if (!empty($item->FromDomain)) {
+                // Conditionally concatenate port if it exists
+                $domain = (string)$item->FromDomain;
+                $port = (string)$item->FromPort;
+                $combinedDomainPort = $domain . (!empty($port) ? ':' . $port : '');
+
+                $result['rows'][] = array(
+                    'id' => (string)$item->getAttributes()['uuid'],
+                    'domainPort' => $combinedDomainPort  // Combined domain and port, conditionally adding port
+                );
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Helper function for search operations in search filter dropdown
+     * It filters entries based on UUIDs provided in the request, which may apply to either
+     * direct (for "reverse") or referenced UUIDs (for "subdomain" and "handle")
+     *
+     * @param \OPNsense\Base\BaseRequest $request The current request instance, used to fetch search parameters.
+     * @param string $modelPath The data model path identifier, pointing to the section of the model being searched.
+     * @param string|null $uuidReferenceSection Specific section name to base the UUID reference on, if any.
+     * @return array Filtered search results.
+     */
+    private function searchActionHelper($request, $modelPath, $uuidReferenceSection = null)
```

Comment:
No need to pass `$request`, it's always in the current object scope as `$this->request` 🙃

Reply:
@AdSchellevis 

Thank you for the feedback. I sat down and completely generalized the function now. I hope I didn't reinvent some wheel.

If you have any more feedback on the existing state, I'd be happy to implement it. But now I can also reuse this function in the future if I have model relation fields again and want to compare UUIDs of referenced items.